presto-spi/src/main/java/io/prestosql/spi/session/PropertyMetadata.java

```diff
@@ -199,4 +201,24 @@ public Object encode(T value)
                 String.class::cast,
                 object -> object);
     }
+
+    public static <T extends Enum<T>> PropertyMetadata<T> enumProperty(String name, String description, Class<T> type, T defaultValue, boolean hidden)
+    {
+        return new PropertyMetadata<T>(
+                name,
+                description,
+                createUnboundedVarcharType(),
+                type,
+                defaultValue,
+                hidden,
+                value -> {
+                    try {
+                        return Enum.valueOf(type, ((String) value).toUpperCase(ENGLISH));
+                    }
+                    catch (IllegalArgumentException e) {
+                        throw new IllegalArgumentException(format("Cannot convert [%s] to enum %s. Possible values are: %s", value, type.getName(), EnumSet.allOf(type)), e);
```

Comment:
PrestoException?

Also, I wouldn’t even mention the fact there’s an enum involved. That’s an internal detail that’s irrelevant for a user specifying this property.